File: ris-live.py
import json

try:
	import websocket
except ImportError:
	raise SystemExit('missing websocket-client')

ws = websocket.WebSocket()
ws.connect('wss://ris-live.ripe.net/v1/ws/?client=bgpexample')

params = {
	'moreSpecific': True,
	'host': 'rrc21',
	'socketOptions': {
		'includeRaw': True
	}
}

ws.send(json.dumps({
		'type': 'ris_subscribe',
		'data': params
}))

for data in ws:
	parsed = json.loads(data)
	print(parsed['type'], parsed['data'])

File: examples/ris-live.js
/*
Subscribe to a RIS Live stream and output every message to the javascript console.

The exact same code will work in Node.js after running 'npm install ws' and including the following line:

const WebSocket = require('ws');
*/
var ws = new WebSocket("wss://ris-live.ripe.net/v1/ws/?client=js-example-1");
var params = {
    moreSpecific: true,
    host: "rrc21",
    socketOptions: {
        includeRaw: true
    } 
};
ws.onmessage = function(event) {
    var message = JSON.parse(event.data);
    console.log(message.type, message.data);
};
ws.onopen = function() {
    ws.send(JSON.stringify({
        type: "ris_subscribe",
        data: params
    }));
};
